test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_distributed_operations.py

```diff
@@ -0,0 +1,134 @@
+from ...integration import DEFAULT_TIMEOUT
+from ...integration.sagemaker.timeout import timeout
+
+MULTI_GPU_INSTANCE = 'ml.p3.16xlarge'
+
+git_config = {'repo': 'https://github.com/huggingface/transformers.git', 'branch': 'v4.4.2'}
+
+# configuration for running training on smdistributed Data Parallel
+distribution = {'smdistributed': {'dataparallel': {'enabled': True}}}
+
+# hyperparameters, which are passed into the training job
+hyperparameters = {
+    'model_name_or_path': 'bert-large-uncased-whole-word-masking',
+    'dataset_name': 'squad',
+    'do_train': True,
+    'do_eval': True,
+    'fp16': True,
+    'per_device_train_batch_size': 4,
+    'per_device_eval_batch_size': 4,
+    'num_train_epochs': 1,
+    'max_seq_length': 384,
+    'max_steps': 100,
+    'pad_to_max_length': True,
+    'doc_stride': 128,
+    'output_dir': '/opt/ml/model'
+}
+# metric definition to extract the results
+metric_definitions = [
+    {"Name": "train_runtime", "Regex": "train_runtime.*=\D*(.*?)$"},
+    {'Name': 'train_samples_per_second', 'Regex': "train_samples_per_second.*=\D*(.*?)$"},
+    {'Name': 'epoch', 'Regex': "epoch.*=\D*(.*?)$"},
+    {'Name': 'f1', 'Regex': "f1.*=\D*(.*?)$"},
+    {'Name': 'exact_match', 'Regex': "exact_match.*=\D*(.*?)$"}]
+
+def validate_or_skip_smdataparallel(ecr_image):
+    if not can_run_smdataparallel(ecr_image):
+        pytest.skip("Data Parallelism is supported on CUDA 11 on PyTorch v1.6 and above")
+
+
+def can_run_smdataparallel(ecr_image):
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    return Version(image_framework_version) in SpecifierSet(">=1.6") and Version(image_cuda_version.strip("cu")) >= Version("110")
+
+
+@pytest.mark.integration("smdataparallel")
+@pytest.mark.model("hf_qa_smdp")
+@pytest.mark.processor("gpu")
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+def test_smdp_question_answering(ecr_image, instance_type, py_version, sagemaker_session, tmpdir):
+    """
+    Tests SM Distributed DataParallel single-node via script mode
+    """
+    validate_or_skip_smdataparallel(ecr_image)
+    instance_count = 1
+    instance_type = "ml.p3.16xlarge"
+    with timeout(minutes=DEFAULT_TIMEOUT):
+        estimator = HuggingFace(
+            entry_point='run_qa.py',
+            source_dir='./examples/question-answering',
+            git_config=git_config,
+            metric_definitions=metric_definitions,
+            role='SageMakerRole',
+            image_uri=ecr_image,
+            instance_count=instance_count,
+            instance_type=instance_type,
+            sagemaker_session=sagemaker_session,
+            transformers_version='4.4.2',
+            pytorch_version='1.6.0',
+            py_version='py36',
+            distribution=distribution,
+            hyperparameters=hyperparameters,
+            debugger_hook_config=False,
+        )
+        estimator.fit()
```

Comment:
Please assign a distinctive job name to allow the training job to be easily identified on SM Console to view its logs/details.

Examples:
- https://github.com/aws/deep-learning-containers/blob/master/test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py#L68-L69
- https://github.com/aws/deep-learning-containers/blob/master/test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smdataparallel.py#L57